utils/dataset_manifest/core.py

```diff
@@ -70,12 +70,34 @@ def __len__(self):
     def resolution(self):
         return (self.width, self.height)
 
-    def validate_key_frame(self, container, video_stream, key_frame):
+    def validate_key_frame(
+        self,
+        container: av.container.InputContainer,
+        video_stream: av.video.stream.VideoStream,
+        key_frame: dict,
+        prev_seek_pts: Optional[int],
+    ) -> tuple[bool, Optional[int]]:
+        """
+        Returns a tuple
+        The first element indicates whether it is possible to seek to the key_frame using its pts (possibly with some decoding after seek).
+        The second element is the pts of the first decoded frame after seeking, if it is possible to seek
+        """
+        container.seek(offset=key_frame["pts"], stream=video_stream)
+
+        seek_pts = None
         for packet in container.demux(video_stream):
             for frame in packet.decode():
+                if seek_pts is None:
+                    seek_pts = frame.pts
+                    # if seek landed on the same frame as previous seek, it is redundant
+                    if prev_seek_pts == seek_pts:
+                        return False, None
+                if frame.pts < key_frame["pts"]:
```

Comment:
Should we perhaps add a limit on how many frames we're willing to iterate over before we give up and raise an `InvalidVideoError`? It seems to me that if we allow unlimited iteration, then there could be pathological inputs that make CVAT do much more work than expected (e.g. decoding the entire video just to get some frames from the middle).